### ISSUE TITLE
PPO2 resultsnot match with the benchmark

### ISSUE THREAD

**Issue by surbhi1944** [hill-a/stable-baselines#618]
I trained the model on Walker2d-v2 using PPO2 with following code (hyperparameters are same as mentioned in the PPO paper). But the graph does not reach to the return as specified in the research paper. 

    import gym
    import random
    import numpy as np
    import os
    import matplotlib.pyplot as plt
    import tensorflow as tf   
    from stable_baselines.common.policies import MlpPolicy
    from stable_baselines.common.vec_env import DummyVecEnv, VecNormalize
    from stable_baselines import PPO2
    from stable_baselines.bench import Monitor
    from stable_baselines.results_plotter import load_results, ts2xy
    from datetime import datetime, timedelta
    import time
    from mpi4py import MPI
    from stable_baselines import logger
    from stable_baselines.common import set_global_seeds
    # Configure things.
    seed=30
    set_global_seeds(seed)
    
    best_mean_reward, nsteps,n_steps = -np.inf, 0,2048
    def callback(_locals, _globals):
        """
        Callback called at each step (for DQN an others) or after n steps (see ACER or PPO2)
        :param _locals: (dict)
        :param _globals: (dict)
        """
        global nsteps, best_mean_reward
        global seed,n_steps
        log_dir = "/home/surabhi/Downloads/github/stable-baselines/results/ppo2/Walker2d-v3/"+str(seed)+"/"
        # Print stats every 1000 calls
        if (nsteps ) % n_steps == 0:
            #print("yo")
            # Evaluate policy training performance
            x, y = ts2xy(load_results(log_dir), 'timesteps')
            if len(x) > 0:
                mean_reward = np.mean(y[-100:])
                print(x[-1], 'timesteps')
                print("Best mean reward: {:.2f} - Last mean reward per episode: {:.2f}".format(best_mean_reward, mean_reward))

                # New best model, you could save the agent here
                if mean_reward > best_mean_reward:
                    best_mean_reward = mean_reward
                    # Example for saving best model
                    #print("Saving new best model")
                    _locals['self'].save(log_dir + 'best_model.pkl')
        nsteps += n_steps
        return True

    path="/home/surabhi/Downloads/github/stable-baselines/results/ppo2/Walker2d-v3/"
    log_dir = path+str(seed)+"/"
    os.makedirs(log_dir, exist_ok=True)
    env = gym.make('Walker2d-v2')
    env.seed(seed)
    env =  Monitor(env, log_dir, allow_early_resets=True)
    env = DummyVecEnv([lambda: env])
    # Automatically normalize the input features
    env = VecNormalize(env, norm_obs=True, norm_reward=False,clip_obs=10.)
    print("man seed  ",np.random.get_state()[1][0])
    n_actions = env.action_space.shape[-1]
    model = PPO2(policy=MlpPolicy, env=env,verbose=1, n_steps=2048, nminibatches=64, lam=0.95, gamma=0.99, noptepochs=10,ent_coef=0.0, learning_rate=3e-4, cliprange=0.2,seed=seed,n_cpu_tf_sess=1)

    start_time=time.time()

    model.learn(total_timesteps=int(1e6),log_dir=log_dir,callback=callback)
    model.save(log_dir + "ppo_walker")
    env.save(os.path.join(log_dir, "vec_normalize.pkl"))

Resulting training graph:
![PPO_Walker2d-v2_seed=30_SmoothWindow=50](https://user-images.githubusercontent.com/51375621/70783714-178fbe00-1daf-11ea-93bf-04e580c7ab5d.png)

While the graph of PPO paper on same parameters is:
![ppo2](https://user-images.githubusercontent.com/51375621/70783696-0e065600-1daf-11ea-80e8-1a729760314f.PNG)

Why results are not matching?

**Comment by araffin** [comment]
Hello,

I see already quite a lot of difference in the two algorithms you want to benchmark.
First, I recommend you to read both our recent [guide](https://stable-baselines.readthedocs.io/en/master/guide/rl_tips.html) (especially the part on evaluation) and the doc of PPO2.

Among the differences:
- you are plotting the training reward curve, you should use a separate env for testing the agent, and use `deterministic=True` when testing it (cf doc)
- the plot mention `Walker-v1` whereas you are training on  `Walker-v2` (I don't know the difference and the version of mujoco may not match neither)
- you are plotting only one seed
- you don't normalize the reward (cf next point on vf clipping)
- PPO2 is not the original PPO (PPO1 is closer to the original), it has value function clipping (which is depending on the reward scale and you can deactivate (cf doc)) and advantage normalization
